internal/server/middlewares.go

```diff
@@ -118,17 +132,17 @@ func requestLogger(lg *zap.Logger) middleware {
 			reader := io.NopCloser(bytes.NewBuffer(buf))
 			req.Body = reader
 
-			body := json.RawMessage(buf)
-			jsonBody, err := json.Marshal(body)
-			if err != nil {
-				lg.Debug("error marshling request body: %v", zap.String("error", err.Error()))
-				http.Error(wr, err.Error(), http.StatusInternalServerError)
-				return
-			}
-
-			clientID, _, _ := req.BasicAuth()
+			if len(buf) > 0 {
+				body := json.RawMessage(buf)
+				jsonBody, err := json.Marshal(body)
```

Comment:
instead of marshalling what is already a json bytes, we can use json.Compact instead.
example
```
if len(buf) > 0 {
	dst := &bytes.Buffer{}
	if err := json.Compact(dst, buf); err != nil {
		lg.Debug("error json compacting request body: %v", zap.String("error", err.Error()))
		http.Error(wr, err.Error(), http.StatusInternalServerError)
		return
	}

	fields = append(fields, zap.String("request_body", dst.String()))
}

ref: https://stackoverflow.com/questions/72609104/how-to-use-golang-to-convert-a-multi-line-json-to-one-line-json
```